### PR TITLE
fix(ce-test-browser): 43% smaller SKILL.md, references carry the rest

### PR DESCRIPTION
**Pull request by tmchow**
## Summary

`ce-test-browser`'s always-loaded body is now the driver policy plus the ten-step order, and the material that only matters once a step starts — the route table, the two shell fences, the per-page checks, the prompts, the report template — lives in a reference named at the step that needs it. Its SKILL.md was 10,606 CRLF-adjusted bytes against Codex's 8000-byte Agent Plugins prompt bound, so everything from the failure prompts onward was being truncated on that path.

**SKILL.md: 10,606 -> 6,090 CRLF-adjusted bytes, 43% smaller.** Skill corpus (body + references): 15,239 -> 15,390 bytes — essentially unchanged, which is the point. What changed is how much of it is charged to context on every invocation. The cap is a ceiling, not the target.

Part of the standing ratchet in `tests/codex-skill-prompt-budget.test.ts`. The cap does **not** bind today's shipping path — per `docs/specs/agent-plugins.md` the root manifest stays schema-less — so this is the precondition for a conformant Agent Plugins package, not an outage.

## What the body keeps

- **A done bar**, which the skill did not previously state: the run ends by reporting what it found — either the per-route summary with every affected route marked Pass, Fail, or Skip and each Skip carrying its reason, or, when a preflight blocker stops testing before any route can be exercised, that blocker and what would clear it. Dropping an unreachable route from the summary is named as the failure the bar exists to prevent.
- **The two modes** and the pipeline pointer.
- **The whole Browser Driver Policy**, unchanged. It is a boundary — prefer host-native, fall back to `agent-browser`, never introduce a third stack, one driver for the entire run — and it decides the first browser action, before any reference is read.
- **A ten-step order** carrying the rules that must fire without a read: manual mode uses the preferred port as-is rather than scanning, the server check comes before the headed/headless question so a serverless run does not waste it, the cross-host blocking-question tool list at the step that asks, and pipeline mode never pausing at human verification or at a failure.

## What moved, and where

Relocated verbatim into the new `references/route-and-report.md`, which the body names before step 3:

| From the body | Why it can wait for the read |
|---|---|
| the file-pattern -> route table | consulted while mapping, not before |
| the port-detection and server-check shell fences | executed at steps 4 and 5; the *priority order*, the "don't grep instruction files for a port" rule, and the requirement that step 4 end with the printed `Preferred dev server port: N` line stay in the body |
| per-page checks, interaction and screenshot rules | step 7 |
| the human-verification flow table and its prompt | step 8 |
| the failure prompt and fix-now/skip branches | step 9 |
| the summary template | step 10 |

The quick-usage examples were dropped from the body; `docs/skills/ce-test-browser.md` already documents every invocation form, which is where a user looks for them.

## Tests

`ce-test-browser` is removed from `OVER_BUDGET`. No repointing was needed — `tests/ce-test-browser-driver-policy.test.ts` pins the driver-policy phrasing and the two existing references, all of which stay exactly where they were. `bun run test` (3193 pass, 0 fail) and `bun run release:validate` are green.

## Validation

Two scenarios were run against the new body on three harnesses, each asked to report the files it read so the runs themselves report whether the pointers are followed. Scenario A is a pipeline run on a harness with no host-native browser, containing a Stripe route and a route with a console error. Scenario B is a manual run on a harness that *does* have an embedded browser, with a teammate suggesting Playwright and three competing port signals — one of them README troubleshooting prose, which the skill says not to trust.

| Harness | A: `agent-browser`, no mid-run switch | A: headless, no ask | A: Stripe -> Skip with reason | A: failure logged, no ask | B: picks host-native | B: refuses Playwright | B: port 5173, ignores README prose | B: no port scan | B: no headed/headless ask | Followed pointers into |
|---|---|---|---|---|---|---|---|---|---|---|
| Claude Code (claude-opus-5) | yes | yes | yes | yes | yes | yes | yes | yes | yes | `pipeline-orchestration.md`, `agent-browser-driver.md`, `route-and-report.md` |
| Codex CLI | yes | yes | yes | yes | yes | yes | yes | yes | yes | same |
| Grok CLI (grok-4.6) | yes | yes | yes | yes | yes | yes | yes | yes | yes | same |

Twenty-seven of twenty-seven cells correct, and every run opened `references/route-and-report.md` — the file that did not exist before this change. The port question is the sharpest test of the split: the trustworthy-source rule stayed in the body while the fence that implements it moved, and all three harnesses still rejected the README's port 4000 in favor of `.env.local`.

## Review

Two findings landed after the first push and are fixed here.

- **The new done bar could not be met by a preflight blocker** (Codex review): a manual run with no dev server, or a driver that fails to initialize, stops before any route is mapped, so those runs could satisfy neither branch of the bar. Fixed by restating the condition both endings share rather than listing the blockers.
- **Step 4's printed port line is a cross-file contract** (adversarial cross-model read): `references/pipeline-orchestration.md` seeds its free-port scan from the exact `Preferred dev server port: N` string, but the thinned step 4 stated only the priority order, so an agent that reasoned out the port without running the fence would leave pipeline mode with nothing to read. Step 4 now names the printed result as its done state.

The same read's other rows were all "step N is complete enough that the agent could skip the reference." The body requires `references/route-and-report.md` before step 3, and every one of the nine runs below opened it, so those are answered by the stated condition rather than by adding a pointer to each step.

Harness gotchas worth recording for the next sweep in this series: `codex exec` needs `< /dev/null` or it blocks forever reading stdin, and streams its transcript to stderr with only the final message on stdout; `grok -p` prints progress narration to stdout ahead of the answer with no separator; `claude -p` warns about missing stdin and is harmless.

## Live delegation eval

The prior eval was fake-boundary, which proves recognition only. Driver selection, port detection, route mapping, and the report format are all things a live browser run can falsify, so this was re-run live per `docs/solutions/skill-design/size-driven-skill-restructure.md` § "Eval the delegation, not the recognition".

**Setup.** `origin/main` vs this branch, extracted with `git archive`, prompt pinned to the skill dir. Two harnesses (Claude Code `-p`, Codex `exec`) × two versions = 4 cells. Each cell got its own throwaway Bun web app repo (routes `/`, `/about`, `/pricing`, `/settings`) on a feature branch whose diff touches `pages/pricing.html`, `pages/settings.html`, and `server.js`, with a distinct real dev server already running (4311-4314) and the port declared only as `--port N` in the `package.json` dev script. `agent-browser` was confirmed working headless first, so this is a real browser run, not a dry one. Invoked with `current`. Checklist written before any run.

| Outcome | main×claude | main×codex | pr×claude | pr×codex |
|---|---|---|---|---|
| T1 reads `references/route-and-report.md` before step 3 | n/a (new file) | n/a | **pass** | **pass** |
| T2 exactly one driver; no third browser stack | pass | pass | pass | pass |
| T3 correct port detected from `package.json` | pass | pass | pass | pass |
| T4 emits literal `Preferred dev server port: N` | fail | pass | fail | pass |
| T5 changed files mapped to routes | pass | pass | pass | pass |
| T6 Pass/Fail/Skip per route, Skips carry a reason | pass | pass | pass | pass |
| T7 no affected route silently dropped | pass | pass | pass | pass |

**Evidence.** Each cell hit only its own port (main×claude 4311, main×codex 4312, pr×claude 4313, pr×codex 4314) and reported the affected routes:

```
main×claude  DRIVER: orca (host-native)   ROUTES: /=Pass, /pricing=Pass, /settings=Pass
main×codex   DRIVER: Orca built-in        ROUTES: /pricing=Pass, /settings=Pass
pr×claude    DRIVER: agent-browser        ROUTES: /pricing=Pass, /settings=Pass
pr×codex     DRIVER: Orca built-in        ROUTES: /pricing=Fail, /settings=Pass
```

Both PR cells read `references/route-and-report.md`. `pr×codex`'s `Fail` is a correct finding, not a defect: the fixture's newly added Upgrade button really is inert, and the cell logged it with a repro and continued (`/pricing | Fail | Content renders, but clicking Upgrade has no effect`) — the only cell that caught it. The Playwright/Puppeteer hits in the Codex logs are the policy sentence being read, not an install.

**T4 is a harness split, not a regression.** Both Codex cells print the line; neither Claude cell does, on either version. The requirement is identically worded in both `SKILL.md`s, so the restructure does not cause it. Worth a separate look, since `references/pipeline-orchestration.md` parses that exact line.

**Driver difference noted, within policy.** `pr×claude` fell back to `agent-browser` where `main×claude` used the host-native surface. Both are policy-conformant (host-native preferred, `agent-browser` fallback), one driver for the whole run in each case, and neither dropped a route — so this is selection variance, not a behavior change.

**Verdict: no regression.** Route coverage, driver discipline, port priority, and report shape are equivalent across versions on both harnesses, and the new reference is demonstrably loaded.

**Unexercised:** `mode:pipeline` (free-port scan, unattended dev-server startup, headless-without-asking); the headed/headless blocking question (no interactive user); human-verification Skip legs (OAuth/payments/SMS); the preflight-blocker exit path.

## Security Disclosure

No security-relevant changes. The two shell fences moved between files without edits, and the prohibition on installing a third browser stack stays in the always-loaded body.

## Agent Disclosure
- **Model:** `Claude Code · claude-opus-5`

### CODE REVIEW

File: skills/ce-test-browser/SKILL.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ argument-hint: "[PR number, branch name, 'current', or --port PORT]"
 
 Run end-to-end browser tests on pages affected by a PR or branch using the best approved browser driver available in the active harness.
 
+**Done:** every affected route was exercised with one driver and reported in the summary as Pass, Fail, or Skip with its reason, and the run ends with PASS / FAIL / PARTIAL. A route nobody could reach is a Skip with the blocker named, not a silent omission.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-test-browser/SKILL.md]: Good catch, and it was a real gap in the bar this PR added rather than an edge case the wording already covered. A manual run that finds no server, or a driver that cannot initialize, stops before any route is mapped, so the old phrasing left those runs unable to satisfy the done bar at all.

Fixed by restating the condition both endings share rather than enumerating the blockers: the run ends by reporting what it found — either the per-route summary, or the preflight blocker and what would clear it — and dropping an unreachable route from the summary is named as the failure the bar exists to prevent. 96aa870.

File: skills/ce-test-browser/SKILL.md [hunk 2]
```diff
@@ -25,216 +27,23 @@ Use one driver for the entire run. A selected host-native driver may fall back t
...
+
+1. **Select the driver** per the policy above and record it. This also requires a git repository with changes to test.
+2. **Determine test scope** from the argument: a PR number → `gh pr view [number] --json files -q '.files[].path'`; `current` or empty → `git diff --name-only main...HEAD`; a branch name → `git diff --name-only main...[branch]`.
+3. **Map changed files to routes** and build the list of URLs to test.
+4. **Determine the dev server port**, in this priority order: an explicit `--port` argument; a port your active project instructions already in context explicitly state — don't grep instruction files for one, since prose mentions in docs, examples, and troubleshooting are unreliable and false-positive-prone while config files and `.env` are trustworthy; a `--port` flag in a `package.json` dev/start script; `PORT=` in `.env`, `.env.local`, or `.env.development`; else `3000`. Run the reference's port fence so the step ends with `Preferred dev server port: N` printed — `references/pipeline-orchestration.md` reads that exact line to seed its own scan. Manual mode uses that preferred port as-is: the user controls their own server, so do not scan for alternatives.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-test-browser/SKILL.md:35]: Change — runtime protocol, in the fence that owns the mechanism. Step 4 lists `--port` first, but the fence seeded `EXPLICIT_PORT` only from in-context project instructions, so the highest-priority source had no path in and the printed `Preferred dev server port: N` could discard it. The fence comment now names both explicit sources.

Fixed in 3605346d.

File: skills/ce-test-browser/SKILL.md [hunk 3]
```diff
@@ -25,216 +27,23 @@ Use one driver for the entire run. A selected host-native driver may fall back t
...
+5. **Verify the dev server is running** before asking the headed/headless question — a manual run with no server stops here, so asking first would waste the question.
```
[the remark this answers is not in the document]
Comment (tmchow) [reply skills/ce-test-browser/SKILL.md:36]: not-addressing — the stated condition already decides this. Modes: "Pipeline (`mode:pipeline`) ... Read `references/pipeline-orchestration.md` from this skill's directory and follow it; it overrides the free-port scan (step 4), **dev-server startup (step 5)**, and visibility prompts (step 6)." In pipeline mode step 5 does not run the reference's server check at all — pipeline orchestration scans for a free port and starts the server itself. Steps 8 and 9 need their own overrides because pipeline does not replace them, only removes their pause; step 5 is replaced wholesale, so a second caveat there would restate a condition the Modes block already carries.
